pylint/checkers/dunder_methods.py

```diff
@@ -118,24 +118,53 @@
 }
 
 
-class DunderCallChecker(BaseChecker):
-    """Check for unnecessary dunder method calls.
+EXTRA_DUNDER_METHODS = [
+    "__new__",
+    "__subclasses__",
+    "__init_subclass__",
+    "__set_name__",
+    "__class_getitem__",
+    "__missing__",
+    "__exit__",
+    "__await__",
+    "__aexit__",
+    "__getnewargs_ex__",
+    "__getnewargs__",
+    "__getstate__",
+    "__setstate__",
+    "__reduce__",
+    "__reduce_ex__",
+    "__post_init__",  # part of dataclasses module
+]
+
+DUNDER_PROPERTIES = [
+    "__class__",
+    "__dict__",
+    "__doc__",
+    "__format__",
+    "__module__",
+    "__sizeof__",
+    "__subclasshook__",
+    "__weakref__",
+]
+
+
+class DunderChecker(BaseChecker):
+    """Checks related to dunder methods.
 
     Docs: https://docs.python.org/3/reference/datamodel.html#basic-customization
-    We exclude __new__, __subclasses__, __init_subclass__, __set_name__,
-    __class_getitem__, __missing__, __exit__, __await__,
-    __aexit__, __getnewargs_ex__, __getnewargs__, __getstate__,
-    __setstate__, __reduce__, __reduce_ex__
-    since these either have no alternative method of being called or
-    have a genuine use case for being called manually.
+
+    For `unnecessary-dunder-call` we exclude dunder names in list
+    `EXTRA_DUNDER_METHODS` since these either have no alternative
+    method of being called or have a genuine use case for being called manually.
```

Comment:
How was it done previously ? It seems the refactor should include removing the old way of doing it to use the new list introduced here.

Reply:
The methods listed in this docstring were never listed in any object. Instead, because the only checker that existed was `unnecessary-dunder-call`, these methods just needed to NOT be included in `DUNDER_METHODS`. BUT with the addition of the `bad-dunder-name` check ,if we don't specifically add these EXTRA methods and check against them, then we would think they are bad-dunder-names when they are not. Does that make sense?

Reply:
Make sense thank you for explaining.